src/yunohost/user.py

```diff
@@ -566,6 +607,225 @@ def user_info(username):
     return result_dict
 
 
+def user_export():
+    """
+    Export users into CSV
+
+    Keyword argument:
+        csv -- CSV file with columns username;firstname;lastname;password;mailbox-quota;mail;mail-alias;mail-forward;groups
+
+    """
+    import csv  # CSV are needed only in this function
+    from io import StringIO
+    with StringIO() as csv_io:
+        writer = csv.DictWriter(csv_io, CSV_FIELDNAMES,
+                                delimiter=';', quotechar='"')
+        writer.writeheader()
+        users = user_list(CSV_FIELDNAMES)['users']
+        for username, user in users.items():
+            user['mail-alias'] = ','.join(user['mail-alias'])
+            user['mail-forward'] = ','.join(user['mail-forward'])
+            user['groups'] = ','.join(user['groups'])
+            writer.writerow(user)
+
+        body = csv_io.getvalue().rstrip()
+    if msettings.get('interface') == 'api':
+        # We return a raw bottle HTTPresponse (instead of serializable data like
+        # list/dict, ...), which is gonna be picked and used directly by moulinette
+        from bottle import HTTPResponse
+        response = HTTPResponse(body=body,
+                        headers={
+                            "Content-Disposition": "attachment; filename=users.csv",
+                            "Content-Type": "text/csv",
+                        })
+        return response
+    else:
+        return body
+
+
+@is_unit_operation()
+def user_import(operation_logger, csvfile, update=False, delete=False):
+    """
+    Import users from CSV
+
+    Keyword argument:
+        csv -- CSV file with columns username;firstname;lastname;password;mailbox_quota;mail;alias;forward;groups
+
+    """
+
+    import csv # CSV are needed only in this function
+    from moulinette.utils.text import random_ascii
+    from yunohost.permission import permission_sync_to_user
+    from yunohost.app import app_ssowatconf
+    # Pre-validate data and prepare what should be done
+    actions = {
+        'created': [],
+        'updated': [],
+        'deleted': []
+    }
+    is_well_formatted = True
+
+    def to_list(str_list):
+        return str_list.split(',') if str_list else []
```

Comment:
Here we probably want to strip spaces, because it's super-easy to edit stuff in a spreadsheet and naively write `foo, bar` instead of `foo,bar`